### PR TITLE
Avoid stale client_vc

### PR DESCRIPTION
**Pull request by shinrich**
The fix in PR #6469 was causing other problems with use-after-free issues being detected by ASAN.  Upon reflection, I think this approach is better to fix the original stale client_vc pointer issue.  Rather than setting do_io_writes with null continuations that could cause the underlying netvc to close itself without signaling to the Session or HttpSM, we should always set the do_io_write continuation to the next live object that should get the network level signal. 

We have run this patch on a couple machines this week without issue.  Our next step is to roll this out to all of our ATS9 machines next week.

### CODE REVIEW

**Comment by shinrich** [comment]
I added the EHTTP_ERROR option to force the HTTP1.x half_close to go away.  So not needed for HTTP/2 since it does not have that problem.  I need to take another pass on this since I am seeing a different crash with this in our production.  I'll try to normalize the error options for the other protocols.  It may be more generally useful to indicate to all the versions of the ProxySessions that the things should go down more agressively.

**Comment by maskit** [comment]
Alright. According to the docs, passing some error code indicates it's an abort. So using it to force half_close sounds fine to me. I guess we've never used it correctly in any of ProxySessions. A right direction would probably be passing an error code on other similar cases even if it is not used inside.

**Comment by shinrich** [comment]
[approve ci clang-analyzer]

**Comment by shinrich** [comment]
Updated this PR, with the commit we added while testing with the ATS9 branch on docs.

File: proxy/http/Http1ClientSession.cc [hunk 1]
```diff
@@ -293,11 +291,7 @@ Http1ClientSession::state_wait_for_close(int event, void *data)
   case VC_EVENT_ACTIVE_TIMEOUT:
   case VC_EVENT_INACTIVITY_TIMEOUT:
     half_close = false;
```
Comment (SolidWallOfCode) [inline proxy/http/Http1ClientSession.cc:293]: This seems redundant - `do_io_close` will set it.

File: proxy/http/Http1ClientSession.cc [hunk 2]
```diff
@@ -373,11 +367,7 @@ Http1ClientSession::state_keep_alive(int event, void *data)
     break;
 
   case VC_EVENT_EOS:
-    this->do_io_close();
-    if (_vc != nullptr) {
-      _vc->do_io_close();
-      _vc = nullptr;
-    }
+    this->do_io_close(EHTTP_ERROR);
```
Comment (SolidWallOfCode) [inline proxy/http/Http1ClientSession.cc:370]: Shouldn't `EOS` not be an error?

Comment (SolidWallOfCode) [reply proxy/http/Http1ClientSession.cc:370]: Explained in person - this is to force full close, never half close. Might be better to not pass an error and explicitly set `half_close = false;` but it's a minor point.

**Pull request merged**